src/wrappers/themis/rust/src/secure_session.rs

```diff
@@ -582,6 +584,13 @@ impl SecureSession {
         unsafe {
             let length =
                 secure_session_send(self.session, message_ptr as *const c_void, message_len);
+            if length == TRANSPORT_FAILURE {
+                let error = self.context.last_error.take().expect("missing error");
+                return Err(Error::from_transport_error(error));
+            }
+            if length == TRANSPORT_OVERFLOW {
+                return Err(Error::with_kind(ErrorKind::BufferTooSmall));
+            }
             if length <= 21 {
```

Comment:
what mean 21?

Reply:
A magic number ✨ Actually that's the maximum numerical value of `themis_status_t` — THEMIS_SSESSION_GET_PUB_FOR_ID_CALLBACK_ERROR.

It is a necessary hack because the current callback API of Secure Session is not able to distinguish between errors and short reads/writes. For example, if `secure_session_receive()` returns 15 then it may mean that 15 bytes of data has been received, or it can also mean a THEMIS_DATA_CORRUPT error. There is an outstanding issue for that somewhere...

I guess this constant deserves a name as well.